hazelcast/src/test/java/com/hazelcast/quorum/atomic/AbstractAtomicQuorumTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.quorum.atomic;
+
+import com.hazelcast.config.AtomicLongConfig;
+import com.hazelcast.config.AtomicReferenceConfig;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.QuorumConfig;
+import com.hazelcast.core.IAtomicLong;
+import com.hazelcast.core.IAtomicReference;
+import com.hazelcast.core.IFunction;
+import com.hazelcast.instance.HazelcastInstanceFactory;
+import com.hazelcast.quorum.PartitionedCluster;
+import com.hazelcast.quorum.QuorumType;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+
+import java.io.Serializable;
+
+import static com.hazelcast.quorum.PartitionedCluster.QUORUM_ID;
+import static com.hazelcast.quorum.QuorumType.READ;
+import static com.hazelcast.quorum.QuorumType.READ_WRITE;
+import static com.hazelcast.quorum.QuorumType.WRITE;
+import static com.hazelcast.test.HazelcastTestSupport.randomString;
+
+public abstract class AbstractAtomicQuorumTest {
```

Comment:
JavaDoc would help maintainability.

Reply:
Yeah, true, each quorum test has such a class. At the end of the PRD I will be trying to pull some sth higher in the hierarchy, and then I'll document the top-level abstract class. Thx for the hint!

Reply:
I hope at least one of us will remember to check it. It took me some time to capture the mapping between configuration and structures (how the naming is used).